src/modals/SendModal.js

```diff
@@ -259,6 +262,18 @@ class SendModal extends Component {
 
   componentDidMount() {
     this.props.sendModalInit();
+    if (typeof window.web3 !== 'undefined') {
+      console.log('web3 browser detected, using.');
+      window.web3.version.getNetwork(function(err, network) {
+        if (err) {
+          console.log(err);
+        }
+        ens = new ENS({
+          provider: window.web3.currentProvider,
+          network: network,
+        });
+      });
+    }
```

Comment:
1. This method is too dependent on Metamask, a user can and should be able to use Balance Manager without Metamask installed
2. Move this logic to `src/handlers/web3.js` and use the `web3Instance` to obtain the `currentProvider`

Reply:
Fixed, please see if the approach feels better now.